CI: azure aks deploy sometimes fail to find the resources group

Recently I've noticed some CI jobs that relies on AKS provisioning failing because the Azure resource group (which is created on-demand before creating the cluster) is not found:

For example, https://github.com/kata-containers/kata-containers/actions/runs/7724334694/job/21074189970?pr=8839 and https://github.com/kata-containers/kata-containers/actions/runs/7722583170/job/21076489139?pr=8974, you will see an error similar to:

```
ERROR: (ResourceGroupNotFound) Resource group 'kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found.
Code: ResourceGroupNotFound
Message: Resource group 'kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found.
{
  "id": "/subscriptions/***/resourceGroups/kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s",
  "location": "eastus2",
  "managedBy": null,
  "name": "kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s",
  "properties": {
    "provisioningState": "Succeeded"
  },
  "tags": null,
  "type": "Microsoft.Resources/resourceGroups"
}
WARNING: The behavior of this command has been altered by the following extension: aks-preview
WARNING: SSH key files '/home/runner/.ssh/id_rsa' and '/home/runner/.ssh/id_rsa.pub' have been generated under ~/.ssh to allow SSH access to the VM. If using machines without permanent storage like Azure Cloud Shell without an attached file share, back up your keys to a safe location
ERROR: (ResourceGroupNotFound) Create or update resource group node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s failed. autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found."
Code: ResourceGroupNotFound
Message: Create or update resource group node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s failed. autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found."
Exception Details:	(ResourceGroupNotFound) Resource group 'node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found.
	Code: ResourceGroupNotFound
	Message: Resource group 'node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found.
Error: Process completed with exit code 1.
```

On https://github.com/kata-containers/kata-containers/blob/main/tests/gha-run-k8s-common.sh#L68 it is **requested** the creation the aforementioned resource group but it doesn't wait the resource to effectively exist. That might be the problem.

--- a/tests/gha-run-k8s-common.sh
+++ b/tests/gha-run-k8s-common.sh
@@ -104,10 +104,12 @@ function get_cluster_credentials() {
 
 function delete_cluster() {
     test_type="${1:-k8s}"
+    local rg
+    rg="$(_print_rg_name ${test_type})"
 
-    az group delete \
-        -g "$(_print_rg_name ${test_type})" \
-        --yes
+    if [ "$(az group exists -g "${rg}")" == "true" ]; then
+        az group delete -g "${rg}" --yes
+    fi
 }
 
 function delete_cluster_kcli() {

